File: java/playwright/src/test/java/dev/selenium/request/LogTest.java
package dev.selenium.request;

import dev.selenium.TestBase;
import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class LogTest extends TestBase {

  @Test
  public void logRequests() {
    List<String> requests = new ArrayList<>();
    context.onRequest(request -> requests.add(request.url()));

    page.navigate("https://demo.playwright.dev/todomvc/");

    Assertions.assertEquals(5, requests.size());
  }
}

File: dotnet/Selenium/Request/LogTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Selenium.Request;

[TestClass]
public class LogTest : TestBase
{
    [TestMethod]
    public async Task LogRequests()
    {
        var requests = new List<string>();
        
        await using var subscription = await Context.Network.OnBeforeRequestSentAsync(req =>
        {
            requests.Add(req.Request.Url);
        });

        // await using var registration = await driver.Network.AddRequestHandlerAsync(
        //     ObserverOptionsBuilder.Defaults(),
        //     req =>
        //     {
        //         requests.Add(req.Url)
        //     });

        await Driver.Navigate().GoToUrlAsync("https://demo.playwright.dev/todomvc/");

        Assert.AreEqual(5, requests.Count);
    }
}